internal/domain/rulebook/dnd5e/calculators/ac_calculator.go

```diff
@@ -0,0 +1,177 @@
+		return 10
+	}
+
+	// Get ability modifiers
+	dexMod := 0
+	wisMod := 0
+	conMod := 0
+
+	if dexScore, exists := char.Attributes[shared.AttributeDexterity]; exists && dexScore != nil {
+		dexMod = dexScore.Bonus
+	}
+	if wisScore, exists := char.Attributes[shared.AttributeWisdom]; exists && wisScore != nil {
+		wisMod = wisScore.Bonus
+	}
+	if conScore, exists := char.Attributes[shared.AttributeConstitution]; exists && conScore != nil {
+		conMod = conScore.Bonus
+	}
+
+	// Check for armor
+	hasArmor := false
+	baseAC := 10
+
+	// Check equipped slots for armor
+	if char.EquippedSlots != nil {
+		for slot, item := range char.EquippedSlots {
+			if item != nil && slot == shared.SlotBody && item.GetEquipmentType() == equipment.EquipmentTypeArmor {
+				hasArmor = true
+				// Try to cast to Armor type to get AC values
+				if armor, ok := item.(*equipment.Armor); ok && armor.ArmorClass != nil {
+					baseAC = armor.ArmorClass.Base
+					// Special handling for leather armor - D&D 5e API may incorrectly set DexBonus to false
+					if item.GetKey() == "leather-armor" || item.GetKey() == "studded-leather-armor" {
+						// Light armor always uses full DEX bonus
+						// Keep dexMod as is
+					} else if !armor.ArmorClass.DexBonus {
+						dexMod = 0 // Heavy armor doesn't use DEX
+					} else if armor.ArmorClass.MaxBonus > 0 {
+						// Limit dex bonus for medium armor
+						if dexMod > armor.ArmorClass.MaxBonus {
+							dexMod = armor.ArmorClass.MaxBonus
+						}
+					}
+				} else {
+					// Fallback for armor without proper AC data
+					baseAC = c.getArmorACFallback(item.GetKey(), &dexMod)
+				}
+				break
+			}
+		}
+	}
+
+	// Get class features
+	classFeatures := []rulebook.CharacterFeature{}
+	if char.Class != nil {
+		classFeatures = features.GetClassFeatures(char.Class.Key, char.Level)
+	}
+
+	// Calculate AC based on armor or unarmored defense
+	var ac int
+	if !hasArmor {
+		// Check for Unarmored Defense
+		if features.HasFeature(classFeatures, "unarmored_defense_monk") {
+			// Monk Unarmored Defense
+			ac = 10 + dexMod + wisMod
+		} else if features.HasFeature(classFeatures, "unarmored_defense_barbarian") {
+			// Barbarian Unarmored Defense
+			ac = 10 + dexMod + conMod
+		} else {
+			// No armor and no unarmored defense
+			ac = 10 + dexMod
+		}
+	} else {
+		// Standard AC with armor
+		ac = baseAC + dexMod
+	}
+
+	// Check for shield (works with any AC calculation including unarmored defense)
+	hasShield := false
+	if char.EquippedSlots != nil {
+		for _, item := range char.EquippedSlots {
+			if item != nil && item.GetKey() == "shield" {
+				hasShield = true
+				break
+			}
+		}
+	}
+
+	if hasShield {
+		ac += 2
+	}
+
+	// Apply Defense fighting style (+1 AC while wearing armor)
+	if hasArmor && char.Features != nil {
+		for _, feature := range char.Features {
+			if feature.Key == "fighting_style" && feature.Metadata != nil {
+				if style, ok := feature.Metadata["style"].(string); ok && style == "defense" {
```

Comment:
avoid casting like this if we can, also defense seems very rulebook specific right?